SECP-service/src/main/java/com/visucius/secp/Controllers/GroupController.java

```diff
@@ -0,0 +1,256 @@
+    {
+        this.groupRepository = groupRepository;
+        this.userRepository = userRepository;
+        this.rolesRepository = rolesRepository;
+        this.permissionsRepository = permissionDAO;
+    }
+
+
+    public Response createGroup(GroupCreateRequest request)
+    {
+
+        String error = validateCreateRequest(request);
+        if(StringUtils.isNoneEmpty(error))
+        {
+                throw new WebApplicationException(
+                    error,
+                    Response.Status.BAD_REQUEST);
+        }
+
+        Group group = new Group(request.name);
+        group.setRoles(getRoles(request.roles));
+        group.setUsers(findUsersWithRoleIDsAndPermissionIDs(request.permissions,request.roles));
+        group.setPermissions(getPermissions(request.permissions));
+        Group createdGroup = groupRepository.save(group);
+
+        return Response.status(Response.Status.CREATED).entity(createdGroup.getId()).build();
+    }
+
+    public Response modifyGroup(GroupModifyRequest request, int id) {
+
+        Optional<Group> groupOptional = getGroup(id);
+        if (!groupOptional.isPresent()) {
+            throw new WebApplicationException(
+                GroupErrorMessages.GROUP_DOES_NOT_EXIST,
+                Response.Status.BAD_REQUEST);
+        }
+
+        Group group = groupOptional.get();
+        String error = validateModifyReqeuest(request, group);
+        if (StringUtils.isNoneEmpty(error)) {
+            throw new WebApplicationException(
+                error,
+                Response.Status.BAD_REQUEST);
+        }
+
+
+        group.addPermissions(getPermissions(request.add_permissions));
+        group.addRoles(getRoles(request.add_roles));
+
+        group.removePermissions(getPermissions(request.remove_permissions));
+        group.removeRoles(getRoles(request.remove_roles));
+
+        group.setUsers(this.findUsersWithRolesAndPermissions(group.getRoles(), group.getPermissions()));
+        Group createdGroup = groupRepository.save(group);
+        return Response.status(Response.Status.CREATED).entity(createdGroup.getId()).build();
+
+
+    }
+
+    private String validateCreateRequest(GroupCreateRequest request) {
+
+        if (!isGroupNameValid(request.name)
+            || !InputValidator.isNameValid(request.name)) {
+            return GroupErrorMessages.GROUP_NAME_INVALID;
+        }
+
+        if(request.permissions.size() < MINIMUM_AMOUNT_OF_PERMISSIONS)
+        {
+            return GroupErrorMessages.GROUP_PERMISSIONS_REQUIRED;
+        }
+
+        String roleError = validateRoles(request.roles);
+        if(StringUtils.isNoneEmpty(roleError))
+            return roleError;
+
+        String permissionsError = validatePermissions(request.permissions);
+        if(StringUtils.isNoneEmpty(permissionsError))
+            return permissionsError;
+
+        return StringUtils.EMPTY;
+    }
+
+    private String validateModifyReqeuest(GroupModifyRequest request, Group currentGroup) {
```

Comment:
This method is doing a lot. Please split it up. It's hard to understand what's going on. Take a look at the comments i made for the GroupModifyRequest class. That would help in splitting this up.